txinput.go

```diff
@@ -49,14 +60,20 @@ func (tx *Tx) addInput(input *Input) {
 // AddP2PKHInputsFromTx will add all Outputs of given previous transaction
 // that match a specific public key to your transaction.
 func (tx *Tx) AddP2PKHInputsFromTx(pvsTx *Tx, matchPK []byte) error {
+	txID := pvsTx.TxID()
 	for i, utxo := range pvsTx.Outputs {
 		utxoPkHASH160, err := utxo.LockingScript.PublicKeyHash()
 		if err != nil {
 			return err
 		}
 
 		if bytes.Equal(utxoPkHASH160, crypto.Hash160(matchPK)) {
-			if err := tx.From(pvsTx.TxID(), uint32(i), utxo.LockingScriptHexString(), utxo.Satoshis); err != nil {
+			if err := tx.From(&UTXO{
```

Comment:
why not just have?
```
TxID: pvsTx.TxID(),
```

Reply:
The `TxID` func hashes, reverses bytes then hex encodes every call. I figured there's no point in doing that work multiple times each iteration since the result never changes, esp since the previous tx could have thousands of outputs.